04-onion-routing.md

```diff
@@ -395,50 +399,51 @@ func NewOnionPacket(paymentPath []*btcec.PublicKey, sessionKey *btcec.PrivateKey
 
 # Packet Forwarding
 
-This specification is limited to `version` `0` packets; however, the structure of
-future versions may change.
+This specification is limited to `version` `0` packets; however, the structure
+of future versions may change.
 Upon receiving a packet, a processing node compares the version byte of the
-packet with its supported versions and aborts the connection if the packet
-specifies a version number it doesn't support.
-For packets with supported version numbers, the processing node then parses the
+packet with its own supported versions and aborts the connection if the packet
+specifies a version number that it doesn't support.
+For packets with supported version numbers, the processing node first parses the
 packet into its individual fields.
 
 The processing node:
   - if the ephemeral public key is NOT on the `secp256k1` curve:
     - MUST abort processing the packet.
     - MUST report a route failure to the origin node.
 
-The processing node then computes the shared secret, as described below, using the private
-key corresponding to its public key and the ephemeral key from the packet.
+Next, the processing node computes the shared secret using the private key
+corresponding to its own public key and the ephemeral key from the packet, as
+described in [Shared Secret](#shared-secret).
 
 The processing node:
-  - if the packet has previously been forwarded or locally redeemed, i.e. packet
-  contains duplicated routing information:
+  - if the packet has previously been forwarded or locally redeemed, i.e. the
+  packet contains duplicate routing information to a previously received packet:
     - if preimage is known:
       - MAY immediately redeem the HTLC using the preimage.
     - otherwise:
       - MUST abort processing and report a route failure.
 
 The above requirements prevent any hop along the route from retrying a payment
-multiple times and attempting to track a payment's progress via traffic
-analysis. Note that this could be accomplished using a log of previous shared
-secrets or HMACs, which could be forgotten once the HTLC would not be accepted
-anyway (i.e. after `outgoing_cltv_value` has passed). Such a log may use a
-probabilistic data structure, but it MUST rate-limit commitments, as necessary,
-in order to constrain the worst-case storage requirements or false positives of
-this log.
-
-The shared secret is used to compute a _mu_-key. The processing node then
-computes the HMAC of the `hops_data` using the _mu_-key. The resulting HMAC is
-compared with the HMAC from the packet.
+multiple times, in an attempt to track a payment's progress via traffic
+analysis. Note that such traffic analysis could be accomplished using a log of
```

Comment:
Technically this is not traffic analysis (which is passive), this is an active probe.  And we're not accomplishing it here, we're preventing it.  So something like:

Note that disabling such probing could be accomplished....

Reply:
fixed